dbt/include/snowflake/macros/adapters.sql

```diff
@@ -124,7 +124,7 @@
   {%- set max_total_results = max_results_per_iter * max_iter -%}
 
   {%- set sql -%}
-    show terse objects in {{ schema_relation }} limit {{ max_results_per_iter }}
+    show terse objects in {{ schema_relation.database }}.{{ schema_relation.schema }} limit {{ max_results_per_iter }}
```

Comment:
Why do we need to do this? Doesn't this go around the quote policy?

Reply:
After looking at the failing test, I think `schema_relation` is expected to be a string of the form `"{database}.{schema}"`. If that's true, these dot paths would produce nothing. That doesn't show up as a jinja error since jinja plows through. It will show up as a database error since it will pass the command `show terse objects in  limit 100000` (no schema at all).